### DISCUSSION TITLE
Not working on Windows 7

### DISCUSSION THREAD

**Discussion by JPhB** [vkbo/novelWriter#1338] [Help] [answered]
I successfully installed the program on Windows 7 but it returns an error message about 0xc00007b. I'm using a 64-bit machine.

![Capture](https://user-images.githubusercontent.com/39348213/216689972-f0436b03-2974-475f-92f2-24a232e79207.JPG)

**Comment by vkbo** [comment, marked as answer]
Windows 7 reached end of life over three years ago. The issue you're seeing there is not with novelWriter itself, but with the embedded Python 3.10, which does not run on Windows 7. See: https://www.python.org/downloads/windows/

![image](https://user-images.githubusercontent.com/1619840/216725961-ff55902f-4e14-4321-a576-5560c4a3b92a.png)

**Reply by JPhB** [reply]
Thank you. As it turns out there is an unofficial Python 3.10 version for Windows 7 [https://github.com/adang1345/PythonWin7](https://github.com/adang1345/PythonWin7).

I installed it, and novelWriter now runs on Windows 7.

**Reply by johnblommers** [reply]
Good. 

The last Windows I used lo these many years hence is Windows 7 and I frankly loved the Aero theme, the look and feel, the transparency effects and the hot task switcher. Every version of Windows that followed Windows 7 seems to get less and less user friendly based on reports I read.

I understand your reason for staying on Windows 7. Power to you.

BTW nowadays I am firmly embedded in the Linux tribe.

**Reply by vkbo** [reply]
@JPhB novelWriter doesn't require Python 3.10, you can just install 3.8 which does support Windows 7. But glad you got it running :)

@johnblommers Since Windows 7 hasn't received security updates in years, I personally wouldn't use it. I see your point though as I too liked Windows 7. Windows 2000 was the first good Windows and XP was all right too. But yeah, I've been on Linux since 2009 when I switched to Ubuntu 9.10. novelWriter is developed on Linux. I have a Windows 10 virtual machine for releasing it for Windows.

**Reply by JPhB** [reply]
`novelWriter doesn't require Python 3.10`: In view of your earlier response I assumed it did. I had Python 3.7 and could have easily upgraded to 3.8.

**Reply by vkbo** [reply]
It's stated numerous places that the minimum requirement is Python 3.7 and Qt 5.10. I just responded to what the error meant.

Python 3.8 is  the last that officially supports Windows 7, and it's good for another [couple of years](https://devguide.python.org/versions/). Python 3.7 reaches end of life this summer. I will probably drop support for Python 3.7 at the end of the year as it is blocking me from making some changes.

**Reply by FrostKnight** [reply]
@vkbo

I wish I had the will to ditch windows before vista came out, it took me till windows 8.1, for the alarm to sound... :/ Anywho,  kudos to you on creating novelwriter, I think I may finally be able to complete something worthwhile thanks to you.

Also, appreciate you making qt webkit an option for building this, since webkitgtk 2.0+ isnt an option. :)

Yours is also the one that is the fastest without crashing on my ultra old Thinkpad and without having insane dependencies.

**Reply by vkbo** [reply]
@FrostKnight novelWriter doesn't need webkit at all, but it is bundled with some of the Qt installers nonetheless. I actively strip it out where I can. The only requirement aside from the main Qt Widget/Core is SVG support to load icons.

**Reply by FrostKnight** [reply]
> @FrostKnight novelWriter doesn't need webkit at all, but it is bundled with some of the Qt installers nonetheless. I actively strip it out where I can. The only requirement aside from the main Qt Widget/Core is SVG support to load icons.

Ah... so its not needed at all. Well that's good to know.

Yours is arguably the fastest one I have found as of yet. Others I have found, have left me wanting and in the case of some, they require bloaty stuff like qt-webengine or other unneeded stuff.

Anywho, thanks for the info.

**Reply by vkbo** [reply]
The QTextEdit widget that forms the core of novelWriter is rather limited, so it makes sense to use a web engine instead for rich text editors when using Python as the main language. It's not necessarily difficult to customise and extend the QTextEdit class, but the performance hit in doing it from Python compared to in C++ is a significant limitation. I have avoided this where I can. It helps a great deal that novelWriter is plain text as there is no need to apply formatting to the text. Iterating through lines is particularly problematic from Python.
